src/dodal/beamlines/p38.py

```diff
@@ -11,23 +15,52 @@
 
 def d11(
     wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> AdAravisDetector:
+) -> SumHDFAravisDetector:
     return device_instantiation(
-        AdAravisDetector,
+        SumHDFAravisDetector,
         "D11",
         "-DI-DCAM-03:",
         wait_for_connection,
         fake_with_ophyd_sim,
+        directory_provider=get_directory_provider(),
     )
 
 
 def d12(
     wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> AdAravisDetector:
+) -> SumHDFAravisDetector:
     return device_instantiation(
-        AdAravisDetector,
+        SumHDFAravisDetector,
         "D12",
         "-DI-DCAM-04:",
         wait_for_connection,
         fake_with_ophyd_sim,
+        directory_provider=get_directory_provider(),
+    )
+
+
+def linkam3(
```

Comment:
On `i22` it was just called `linkam`?

I'm worried about the potential confusion from a method called `linkam3` returning a device pointing at PVs prefixed `LINKM-02`.